File: backend/app/schemas/terminal.py
from pydantic import BaseModel, IPvAnyAddress
from typing import Optional
from datetime import datetime

class TerminalBase(BaseModel):
    number: int
    name: str
    ip_address: str
    port: int = 4370
    is_active: bool = True
    is_main: bool = False

class TerminalCreate(TerminalBase):
    pass

class TerminalUpdate(BaseModel):
    number: Optional[int] = None
    name: Optional[str] = None
    ip_address: Optional[str] = None
    port: Optional[int] = None
    is_active: Optional[bool] = None
    is_main: Optional[bool] = None

class TerminalResponse(TerminalBase):
    id: int
    last_sync_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True 

File: frontend/src/types/terminal.ts
export interface Terminal {
    id: number;
    number: number;
    name: string;
    ip_address: string;
    port: number;
    is_active: boolean;
    is_main: boolean;
    last_sync_at: string | null;
    created_at: string;
    updated_at: string;
}

export interface CreateTerminalData {
    number: number;
    name: string;
    ip_address: string;
    port: number;
    is_active: boolean;
    is_main: boolean;
}

export interface UpdateTerminalData {
    name?: string;
    ip_address?: string;
    port?: number;
    is_active?: boolean;
    is_main?: boolean;
}

export interface TerminalFormData {
    number: number;
    name: string;
    ip_address: string;
    port: number;
    is_active: boolean;
    is_main: boolean;
}

export interface SyncResult {
  status: string;
  message: string;
  stats: {
    terminal_id: number;
    terminal_name: string;
    total_employees: number;
    synced_employees: number;
    sync_details: {
      status: string;
      message: string;
      updated_count: number;
    };
    changes: SyncChange[];
  };
}

export interface SyncChange {
  type: 'add' | 'update';
  employee: string;
  enroll_number: string;
  message?: string;
  changes?: string[];
} 